obsei/analyzer/ner_analyzer.py

```diff
@@ -78,8 +80,9 @@ def analyze_input(
         analyzer_config: Optional[BaseAnalyzerConfig] = None,
         **kwargs
     ) -> List[TextPayload]:
-        if analyzer_config is None:
-            raise ValueError("analyzer_config can't be None")
+        # if analyzer_config is None:
+        #     raise ValueError("analyzer_config can't be None")
+        ner_batch_size = 1
```

Comment:
Interesting. Thanks for pointing it out.

Ideally `batch_size` should be part of Analyzer. Hence in this case `BaseAnalyzer` is correct place for this.
Let's move`batch_size` from `analyzer_confg` to `BaseAnalyzer` and correct this anomaly.

Reply:
Isnt `BaseAnalyzerConfig` is inherited by other AnalyzerConfig class meant for configurations like batch size or labels. 
By moving batch_size to `BaseAnalyzer` will be more unclean , 

I suggest that we can add a BaseAnalyzerConfig as a class variable with batch size 1 in NerAnalyzer , this will be a cleaner way. 

```
class NERAnalyzer(BaseAnalyzer):
    _pipeline: Pipeline = PrivateAttr()
    _max_length: int = PrivateAttr()
    TYPE: str = "NER"
    model_name_or_path: str
    tokenizer_name: Optional[str] = None
    grouped_entities: Optional[bool] = True

    ner_config : BaseAnalyzerConfig = BaseAnalyzerConfig(batch_size=1)   # This line

```

and use this ner config in place of `analyzer_config`  in `analyze_input`

Reply:
batch_size is more of infra property instead of configuration ie labels, hence better suited near model name and other parameters. For example if transformers model inherent provide batching then Obsei might skip batching at all. Also value of batch size will be depend on where we are running model via analyzer ie system of high or low memory?

Let's assume ClassificationAnalyzer running on CPU and GPU will use different batch_size but same analyzer_config.